controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -195,12 +196,94 @@ function fetch($request, $actionArgs = array()) {
 		$userDao = DAORegistry::getDAO('UserDAO');
 		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
 		$participantOptions = array();
+
 		foreach ($assignments->toArray() as $assignment) {
+			$userGroup = $userGroupDao->getById($assignment->getUserGroupId());
 			$participantOptions[] = array(
 				'user' => $userDao->getById($assignment->getUserId()),
-				'userGroup' => $userGroupDao->getById($assignment->getUserGroupId()),
+				'userGroup' => $userGroup->getLocalizedName(),
 				'isParticipant' => in_array($assignment->getUserId(), $currentParticipants),
+				'userId' => $assignment->getUserId(),
+				'roleId' => $userGroup->getRoleId(),
+				'reviewOpen' => false,
 			);
+
+			# Check the assignment of the current user
+			if ($user->getId() == $assignment->getUserId()){ $userAssignment = $userGroup->getRoleId(); }
+		}
+
+		# In review stage, add reviewers as possible participants
+		if ($query->getStageId() == WORKFLOW_STAGE_ID_EXTERNAL_REVIEW) {
+			$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO');
+			$reviewAssignments = $reviewAssignmentDao->getBySubmissionId($query->getAssocId());
+
+			foreach ($reviewAssignments as $reviewAssignment) {				
+
+				# Check if current user is a reviewer, and if the used method is blind
+				if ($user->getId() == $reviewAssignment->getReviewerId()){ 
+					$userAssignment = ROLE_ID_REVIEWER;
+					if ($reviewAssignment->getReviewMethod() != SUBMISSION_REVIEW_METHOD_OPEN) { $isBlindReviewer = true; }
+				}
+
+				# Check if review is open and confirmed
+				$openReview = false;
+				if ($reviewAssignment->getReviewMethod() == SUBMISSION_REVIEW_METHOD_OPEN && $reviewAssignment->getDateConfirmed()) { $openReview = true; }
+
+				# ALEC! userGroup is now hard coded, because reviewAssignment does not have userGroupId!
```

Comment:
I am thinking of a situation where a journal has some specially named reviewer role that has the reviewer capabilites, but is not the same thing as the reviewer. 

I do not have a use case, but it would make sense the assignment information would include the role that was used in the assignment. Just like it does in the stage assignment. I mean if a user has both the "usual" reviewer role and the "special" reviwer role, then now there would be no way of telling which role was used to assign the review, right?

Reply:
As far as I'm concerned, the `review_assignment` entry will always and only refer to reviewers. If in the distant future we change this, we'll be up for a pretty comprehensive review of the code. Practically speaking I'm not aware of a variety of reviewer roles being used, beyond the "internal" and "external" reviewers we have in OMP.